src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/Telemetry/TelemetryReporter.cs

```diff
@@ -16,6 +17,15 @@ namespace Microsoft.VisualStudio.Razor.Telemetry;
 
 internal abstract class TelemetryReporter : ITelemetryReporter
 {
+    private const string CodeAnalysisNamespace = nameof(Microsoft) + "." + nameof(CodeAnalysis);
```

Comment:
I think we're going to have a half dozen of these where we want to report the caller; is there a unittest to write to make sure we are detecting/rewriting info as we expect?

Reply:
Unfortunately I didn't see a way to check the properties being set here. @pieandcakes do you know?

Reply:
That I can add, but if we can find a way to assert the fault behavior I would prefer that. If not testing `GetModifiedFaultParameters` is still a good test.

Reply:
We don't process the bucket parameters until PostEvent is called and then it processes them and adds them to the properties. I don't think you'd have a way to validate that the values are correct/not correct

Reply:
Got it. Should the `Module` be something like `Microsoft.VisualStudio.LanguageServices.Razor.Test.dll` or should it be the containing type full name?

Reply:
It should be FileNameNoExtensions. The code that does it today is here:

https://devdiv.visualstudio.com/DevDiv/_git/VSTelemetryAPI?path=/src/Microsoft.VisualStudio.Telemetry/WindowsErrorReporting/WatsonReportExceptionProcessing.cs&version=GBmain&line=463&lineEnd=464&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents